package.json

```diff
@@ -21,7 +21,7 @@
     "conformance": "./run_conformance_tests.sh",
     "check": "gts check",
     "clean": "gts clean",
-    "compile": "rm -rf ./build && tsc -p .",
+    "compile": "rm -rf ./build && tsc -p . && api-extractor run --local --verbose",
```

Comment:
While it's great to add the API extractor in our scripts, I don't suggest we make the tool run in the TS compilation script.

Moreover, this looks like a copy of `&& npm run docs`.

We already have a `predocs` script that runs compile.

Reply:
the reason I didn't make it `&&  npm run docs` is because we have a `predocs` that runs `compile` so it would have created a cycle...

Reply:
Right. However, we're repeating ourselves though with `api-extractor run --local --verbose`.

There are a few ways to avoid this. For example, creating a separate command like `api-extractor` and running the command there.

However, I'm not sure we want to run the doc extractor when we compile.

For example, it looks like this breaks our watch command:

```
npm run watch

> @google-cloud/functions-framework@1.10.1 watch
> npm run compile -- --watch


> @google-cloud/functions-framework@1.10.1 compile
> rm -rf ./build && tsc -p . && api-extractor run --local --verbose "--watch"


api-extractor 7.18.16  - https://api-extractor.com/

usage: api-extractor run [-h] [-c FILE] [-l] [-v] [--diagnostics]
                         [--typescript-compiler-folder PATH]
                         
api-extractor run: error: Unrecognized arguments: --watch.
```

Reply:
hmm yeah, I think the key think is I want to make sure we run api-extractor before we publish (so in prepare). I will try to refactor this a bit

Reply:
OK I refactored the scripts a bit:

1. removed `predocs`
2. replaced `clean` with the first part of `compile`
3. created a new `build` that runs `clean`, `compile`, `docs`
4. made `prepare` and alias for `build`